src/structlog/stdlib.py

```diff
@@ -444,3 +444,23 @@ def wrap_for_formatter(logger, name, event_dict):
         configuration.
         """
         return (event_dict,), {"extra": {"_logger": logger, "_name": name}}
+
+    @staticmethod
+    def get_logger_name_and_msg(record):
+        # Both attached by wrap_for_formatter
+        logger = record._logger if hasattr(record, '_logger') else None
```

Comment:
JFTR, `hasattr()` is an [unfortunate Python idiom](https://hynek.me/articles/hasattr/) and `getattr(record, "_logger", None)` does *exactly* what you want here; just faster and better. :)